src/mock_etcd.rs

```diff
@@ -609,4 +893,147 @@ mod test {
             assert_eq!(resp.count(), 0);
         }));
     }
+
+    fn e2e_watch_test() {
+        let mut etcd_server = MockEtcdServer::new();
+        etcd_server.start();
+
+        smol::future::block_on(Compat::new(async {
```

Comment:
How do you know this function is finished when you run it in detach mode?

Reply:
You mean `smol::spawn(async move ` this function? We do not check if this function is finished, we are just checking if the watch responses are correct.